server/src/main/java/com/saralein/server/middleware/FileMiddleware.java

```diff
@@ -43,62 +36,40 @@ public Middleware apply(Callable callable) {
 
     @Override
     public Response call(Request request) {
-        String resource = fileHelper.createAbsolutePath(request.getUri());
-        Path resourcePath = Paths.get(resource);
-
-        if (resourceExists(resourcePath)) {
-            return getResponse(resourcePath, request);
+        if (fileExists(request)) {
+            return getFileResponse(request);
         } else {
             return next.call(request);
         }
     }
 
-    private Response getResponse(Path resource, Request request) {
+    private Response getFileResponse(Request request) {
         if (!isAcceptedMethod(request.getMethod())) {
             return accessNotAllowed();
         }
 
-        if (isDirectory(resource)) {
-            return getDirectoryResponse(request);
-        }
-
-        return getFileResponse(request);
-    }
-
-    private Response getDirectoryResponse(Request request) {
         try {
-            return directoryHandler.handle(request);
-        } catch (IOException e) {
+            return fileHandler.handle(request);
+        } catch (Exception e) {
             return serverError();
         }
     }
 
-    private Response getFileResponse(Request request) {
-        try {
-            return fileHandler.handle(request);
-        } catch (IOException e) {
-            return serverError();
-        }
+    private boolean fileExists(Request request) {
```

Comment:
This method  and a similar method are in most of these file related middlewares, perhaps they are better off on the file handler?

Reply:
It looks like this has already been addressed in the next PR